src/perftest_communication.c

```diff
@@ -906,29 +909,30 @@ int rdma_client_connect(struct pingpong_context *ctx,struct perftest_parameters
 {
 	char *service;
 	int temp,num_of_retry= NUM_OF_RETRIES;
-	struct sockaddr_in sin, source_sin;
+	struct sockaddr_storage source_sin;
+	struct sockaddr *sin;
 	struct sockaddr *source_ptr = NULL;
 	struct addrinfo *res;
```

Comment:
Now I get to know why the initial author kept a copy for both source and dest addresses respectively, `res` is reused and freed right after the address in it is copied out.
suggest:
```c
	struct sockaddr_storage sin, source_sin;
	struct sockaddr *sin_ptr = NULL, *source_ptr = NULL;
	struct addrinfo *res;
```

Reply:
can you please explain and elaborate more about this comment, specially about the sin_ptr and its usage?

Reply:
Looked at the code again, `sin_ptr` is unnecessary, you can safely cast `struct sockaddr_storage source_sin` to `struct sockaddr *`.  `struct sockaddr *source_ptr = NULL;` is needed to pass to `rdma_resolve_addr` when the source_ip is not supplied.

The target of these addresses is to get cm_id from [rdma_resolve_addr(3)](https://linux.die.net/man/3/rdma_resolve_addr). `rdma_resolve_addr` requires the source (optional) and dest addresses to be `struct sockaddr *`.
In the old code, `struct sockaddr_in sin, source_sin;` are used as the storage for the dest and source addresses respectively that are copied from `struct addrinfo *res`. 

revise my suggestion to:
```c
	struct sockaddr_storage sin, source_sin;
	struct sockaddr *source_ptr = NULL;
	struct addrinfo *res;
```

`sin` and `source_sin` are used to store the addresses from `struct addrinfo *res`. When `rdma_resolve_addr` is called, cast `sin` to `struct sockaddr *`, `source_ptr` is already a `struct sockaddr *` pointer to either NULL or `&source_sin`